spec/index.bs

```diff
@@ -596,24 +596,36 @@ Issue(gpuweb/gpuweb#296): Consider merging all read-only usages.
 Textures may consist of separate [=mipmap levels=] and [=array layers=],
 which can be used differently at any given time.
 Each such <dfn dfn>subresource</dfn> is uniquely identified by a
-[=texture=], [=mipmap level=], and
-(for {{GPUTextureDimension/2d}} textures only) [=array layer=].
+[=texture=], [=mipmap level=],
+(for {{GPUTextureDimension/2d}} textures only) [=array layer=],
+and [=aspect=].
```

Comment:
I don't think this would be correct. In WebGPU today we don't support separate usage tracking for different aspects of a depth-stencil texture.

Reply:
Don't we support it with `readOnlyDepth` and `readOnlyStencil`?

Reply:
Yeah. we do have depth-only and stencil-only in WebGPUTextureAspect, see https://gpuweb.github.io/gpuweb/#enumdef-gputextureaspect.

Reply:
we do have them, but we don't support a case where you have both depth and stencil, and their read-only-ness (or usage, in general) is different, today

Reply:
I am not 100% sure about whether this use fall into this requirement: when we draw shadows, and we need to read data from a few depth-only subresources, and write into the same (invalid) or different depth-only (or depth/stencil) subresources?

Reply:
Today, mutable depth and immutable stencil (and visa versa) is not allowed in WebGPU. They need to either be both read-only, or both mutable.

In terms of the spec text, we could go one of two ways:
  1. say that "aspect" makes up a different subresource, like the text you have in the PR, but then have no real place in the API where the users *could* make them different. I think this is a bit confusing to the reader of the spec: what's the point of saying these subresources are different if there is no way to use them differently?..
  2. say that depth/stencil are the same subresource, like what we have today. Once it becomes possible to use them differently, we can make that spec change.

Reply:
Today, mutable depth and immutable stencil (and visa versa) is not allowed in WebGPU. They need to either be both read-only, or both mutable.

In terms of the spec text, we could go one of two ways:
  1. say that "aspect" makes up a different subresource, like the text you have in the PR, but then have no real place in the API where the users *could* make them different. I think this is a bit confusing to the reader of the spec: what's the point of saying these subresources are different if there is no way to use them differently?..
  2. say that depth/stencil are the same subresource, like what we have today. Once it becomes possible to use them differently, we can make that spec change.

Reply:
I didn't see any statement about "mutable depth and immutable stencil (and visa versa) is not allowed in WebGPU", but at native side, I remember that we can have mutable depth and immutable stencil (and visa versa) at native side (say in D3D12 and Vulkan). We can set different read/write states for different plane slices (aspect) via transition barrier.

Reply:
Hmm, I should have asked for your (Dzmitry's) review on the tests or test plan where we started discussing this. In my mental model, the aspects are different subresources. I believe we even started tracking them this way in Dawn and I'm pretty sure that actually has some effect (@austinEng can you confirm?)

Reply:
See the "Proposal" in https://github.com/gpuweb/gpuweb/issues/514#issuecomment-628925960
The consensus was that the initial read-only can't really be applied separately to depth and stencil, but we leave the door open to enable it some day, possibly via an extension.
The spec text in this area is missing details in general.

Reply:
Ah right, forgot we hadn't resolved that. It sounds like we were moving toward allowing it by default if possible, though.

Tracking depth and stencil as separate subresources would be required to enable this later, would it just be better to track them separately from the start?

Reply:
Yes, I agree. Looking forward, specifying them to be tracked separately is better.